src/test/java/calculator/NumberExtractor.java

```diff
@@ -0,0 +1,48 @@
+package calculator;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+public class NumberExtractor {
+
+    private static final String DEFAULT_DELIMITER = "[,:]";
+
+    private static final String CUSTOM_DELIMITER_PREFIX = "//";
+    private static final String CUSTOM_DELIMITER_SUFFIX = "\n";
+
+    public String extractDelimiter(final String text) {
+        if (text.startsWith(CUSTOM_DELIMITER_PREFIX)) {
+            return text.split(CUSTOM_DELIMITER_SUFFIX)[0].substring(2);
+        }
+
+        return DEFAULT_DELIMITER;
+    }
+
+    public String extractNumberText(final String text) {
+        if (text.startsWith("//")) {
+            return text.split("\n")[1];
+        }
+
+        return text;
+    }
+
+    public void validateNumber(final Integer number) {
+        if (number < 0) {
+            throw new RuntimeException("Negative numbers are not allowed");
+        }
+    }
+
+    public Numbers extract(final String text) {
+        if (text == null || text.isEmpty()) {
+            return new Numbers(List.of());
+        }
+
+        String delimiter = extractDelimiter(text);
+        String numberText = extractNumberText(text);
+
+        return new Numbers(Stream.of(numberText.split(delimiter))
+            .map(Integer::parseInt)
+            .peek(this::validateNumber)
+            .toList());
```

Comment:
아래와 같이 변경해볼 수 있을 것 같아요. 😉 

```suggestion
        return Stream.of(numberText.split(delimiter))
            .map(Integer::parseInt)
            .peek(this::validateNumber)
            .collect(Collectors.collectingAndThen(Collectors.toList(), Numbers::new));
```

Reply:
안녕하세요 @Rok93 리뷰주셔서 감사합니다. 
제안 주신 방식이 Stream API 하나로 구성하여 깔끔하고 가독성면에서 더 괜찮아보입니다! 혹시 제안 주신 내용이 다른 이유도 있을지 궁금합니다 : )

Reply:
생성자 파라미터로 Stream 구문을 통으로 주입하고있어서 가독성을 해친다고 생각했어요. 
저라면 기존의 방식대로 작성했다면 별도로 Stream 구문의 연산 결과를 변수로 추출하고 추출한 변수를 생성자 파라미터로 전달했을 것 같아요. 🙂

단순히 가독성의 얘기였어요.